hbshelpers/codegen/TransformFacetsCall.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Responsible for generating transformFacets function calls
+ */
+class TransformFacetsCall {
+  /**
+   * Generates a transformFacets function based on the fields config
+   * 
+   * @param {Object} fields filterOptions keyed by fieldId
+   */
+  static generate (fields) {
```

Comment:
My main question is whether or not it would be beneficial to create some data models for these types